### PR TITLE
Avoid concurrency by canceling triggered worklfows before running new ones

### LINKED ISSUE #158: Add concurrency to our workflows
Ideally the previous workflow is cancelled when the newer workflow is already running. This can be achieved with [workflow concurrency](https://docs.github.com/en/actions/using-jobs/using-concurrency). Adding this to every workflow should be sufficient:

```yaml
concurrency:
  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}-docker
  cancel-in-progress: ${{ github.ref_name != github.event.repository.default_branch }}
```

Note that the suffix of the group is always specific to the workflow to avoid that different workflows cancel each other in the same run.

### PR DESCRIPTION
**Pull request by MichaelKora**
Fixes #158

### CODE REVIEW

File: .github/workflows/bump-version-release.yaml [hunk 1]
```diff
@@ -44,10 +44,13 @@ on:
         description: "The old version in your `.bumpversion.cfg` file."
         value: ${{ jobs.release.outputs.old-version }}
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}-bump-version-release
+  cancel-in-progress: true
+
```
Comment (yannick-roeder) [inline .github/workflows/bump-version-release.yaml:50]: We do not want workflows that create tags or releases to every be canceled while they are running

File: .github/workflows/docker-build-and-publish.yaml [hunk 2]
```diff
@@ -67,11 +68,14 @@ on:
         description: "GitHub token."
         required: false
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}-docker
+  cancel-in-progress: true
```
Comment (yannick-roeder) [inline .github/workflows/docker-build-and-publish.yaml:73]: Why did you change the `cancel-in-progress` everywhere? The idea is to only cancel the workflow when we are not on the default branch, so that we still let every pipeline run for every merge to main (even if they are running at the same time)

File: .github/workflows/java-gradle-release.yaml [hunk 3]
```diff
@@ -55,6 +56,10 @@ on:
         description: "Bumped version of your project."
         value: ${{ jobs.release.outputs.release-version }}
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}-java-gradle-release
+  cancel-in-progress: true
+
```
Comment (yannick-roeder) [inline .github/workflows/java-gradle-release.yaml:62]: We do not want workflows that create tags or releases to every be canceled while they are running

File: .github/workflows/helm-gke-deploy.yaml [hunk 4]
```diff
@@ -70,11 +71,14 @@ on:
         description: "GKE cluster for authentication"
         required: true
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}-helm-deploy
```
Comment (yannick-roeder) [inline .github/workflows/helm-gke-deploy.yaml:75]: For the Kustomize you included the `gke` in the suffix, but for Helm you don't? Please be consistent about your suffixes

File: .github/workflows/python-poetry-release.yaml [hunk 5]
```diff
@@ -57,6 +58,10 @@ on:
         description: "The old version of the package."
         value: ${{ jobs.create-release.outputs.old-version }}
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}-python-poetry-release
+  cancel-in-progress: true
+
```
Comment (yannick-roeder) [inline .github/workflows/python-poetry-release.yaml:64]: We do not want workflows that create tags or releases to every be canceled while they are running

File: .github/workflows/release-tag-versions.yaml [hunk 6]
```diff
@@ -31,10 +32,13 @@ on:
         description: "The GitHub token for committing the changes."
         required: true
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}-release-version
+  cancel-in-progress: true
+
```
Comment (yannick-roeder) [inline .github/workflows/release-tag-versions.yaml:38]: We do not want workflows that create tags or releases to every be canceled while they are running

**Review by yannick-roeder — CHANGES_REQUESTED** [review]

File: .github/workflows/helm-gke-deploy.yaml (hunk 4 above)
Comment (MichaelKora) [reply .github/workflows/helm-gke-deploy.yaml:75]: my bas.. i missed those two. But the idea is to give the nave of the workflow

**Review by yannick-roeder — APPROVED** [review]
